### PR TITLE
Remove differentiable cycle closure

### PR DESCRIPTION
**Pull request by maxentile**
Reverts https://github.com/proteneer/timemachine/pull/399 (and partially reverts https://github.com/proteneer/timemachine/pull/449):
* the problem it addressed is no longer relevant to timemachine (no longer using networks of comparisons that contain cycles)
* want to remove a dependency that has recently become inconvenient (a sub-dependency of cvxpylayers mishandles its `tests/` directory, so `import tests` can produce a name collision https://github.com/proteneer/timemachine/pull/522)
* if we later want to restore this functionality using a different dependency, https://github.com/google/jaxopt may be more suitable

### CODE REVIEW

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 1]
```diff
@@ -138,21 +137,11 @@ def loss_fxn(ff_params, batch: List[Tuple[RelativeFreeEnergy, RBFEModel]]):
...
-    try:
-        layer = construct_mle_layer(len(index), indices)
-        corrected_dg = layer(jnp.asarray(preds))
-        cycle_corrected_rbfes = corrected_dg[ind_r] - corrected_dg[ind_l]
-    except DisconnectedEdgesError:
-        # Provided non connected graph, use the ddGs directly
-        cycle_corrected_rbfes = jnp.asarray(preds)
     labels = jnp.asarray([rfe.label for rfe, _ in batch])
-    loss = pseudo_huber_loss(cycle_corrected_rbfes - labels)
+    loss = pseudo_huber_loss(preds - labels)
```
Comment (badisa) [inline examples/hif2a/fit_to_multiple_rbfes.py:141]: Does this need `psuedo_huber_loss(jnp.asarray(preds) - labels)`?

Comment (maxentile) [reply examples/hif2a/fit_to_multiple_rbfes.py]: Yes -- thanks for catching. Fixed in https://github.com/proteneer/timemachine/pull/523/commits/67b5c988c9dba4f4902eeb7aace3024ecc6bb4ef

**Review by badisa — APPROVED** [review]
LGTM

**Review by proteneer — APPROVED** [review]
lgtm - looking forward to bringing it back one day via jaxopt
